megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectFireHandler.java

```diff
@@ -145,8 +145,10 @@ public boolean handle(IGame.Phase phase, Vector<Report> vPhaseReport) {
         final Vector<Integer> spottersBefore = aaa.getSpotterIds();
         Coords targetPos = target.getPosition();
         final int playerId = aaa.getPlayerId();
-        boolean isFlak = (target instanceof VTOL) || target.isAero();
-        boolean asfFlak = target.isAero();
+        boolean targetIsEntity = target.getTargetType() == Targetable.TYPE_ENTITY;
+        boolean targetIsAirborneVTOL = targetIsEntity && ((Entity) target).isAirborneVTOLorWIGE();
```

Comment:
Why not do these two steps at once? I understand splitting it off isFlak because of the length, but otherwise.

Reply:
Mostly for ease of debugging. The more complicated a piece of boolean logic is, the less readable it is. This allows me to make the logic for "targetIsAirborneVTOL" human-readable (more or less).